src/api-v1.js

```diff
@@ -70,100 +70,106 @@ api.declare({
     // Intentionally vague because we don't want to reveafollowRedirects
     // our configuration too much
     let msg;
+    let code = 503; // default is that something is temporarily wrong
     switch (err.code) {
-      case 'HTTPError':
-        msg = 'HTTP Error while trying to resolve redirects';
-        break;
       case 'DoesNotMatchPatterns':
         msg = 'Input URL does not match whitelist';
+        code = 403; // forbidden
         break;
       case 'InsecureURL':
         msg = 'Refusing to follow a non-SSL redirect';
+        code = 403; // forbidden
+        break;
+      case 'HTTPError':
+        msg = 'HTTP Error while trying to resolve redirects';
         break;
       default:
         msg = 'Input URL failed validation for unknown reason';
         break;
     }
-    return res.status(400).json({
+    return res.status(code).json({
       msg: msg,
+      code: code,
     });
   }
 
-  if (service.toLowerCase() === 's3') {
-    if (this.s3backends[region]) {
-      let backend = this.s3backends[region];
-
-      let maxWaitForCachedCopy = this.maxWaitForCachedCopy;
-
-      let startTime = new Date();
-      let x = 0;
-      while (new Date() - startTime < maxWaitForCachedCopy) {
-        debug(`Check ${x++} of ${url}`);
+  // This is the ID that we need to find a backend for
+  let incomingId = `${service}_${region}`;
+  
+  // Let's pick which backend to use.  In this case, we're looking to find the
+  // only backend known that matches the potential id
+  let backends = this.cacheManagers.filter(x => x.id === incomingId);
+  if (backends.length > 1) {
+    debug('API server is misconfigured and has more than one cachemanager with id, crashing' + incomingId);
+    // Because this should never ever happen
+    process.exit(-1);
+  } else if (backends.length === 0) {
+    debug(`${incomingId} is not known`);
+    return res.status(404).json({
+      msg: `${incomingId} is not known`,
+    });
+  } else {
```

Comment:
I will change to this, but in general, I find that the explicit else is easier to understand the code as it's a nice hint when reading.